libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -3142,9 +3175,9 @@ data ListFn fn args res where
     ) =>
     fn '[a] b ->
     ListFn fn '[[a]] b
-  ListSize :: ListFn fn '[[a]] Int
+  ListSize :: ListFn fn '[[a]] Size
```

Comment:
This was an oversight on my part. I should have removed it. I am surprised though, that leaves only FoldMap as the only operator on Lists.  I would have thought perhaps Append, and definitely NullList.  
I added a TODO about adding those two at some time.

Reply:
NullList can be done with `length_ xs ==. 0` and append is really tricky because that would require talking about the order of elements and distributing foldMap over the two lists.

It's possible it can be done, but I don't know if it would be at the cost of the interaction with foldMap. On the other hand, one of the lists is known when doing propagate spec so it might be possible that you never need to distribute and only need to keep a prefix and a suffix in the spec.

The first thing I would try is cons however, that will tell us if it can be done easily.